Dockerfile.windows

```diff
@@ -57,12 +57,18 @@ RUN Write-Host ('Installing Visual C++ Redistributable Package'); `
 
 #
 # Install OpenSSL
+# This must be done after installing Visual Studio
 #
 ADD https://slproweb.com/download/Win64OpenSSL-1_1_1L.msi /local/Win64OpenSSL-1_1_1L.msi
 
-RUN Start-Process msiexec.exe `
-    -ArgumentList '/i C:\local\Win64OpenSSL-1_1_1L.msi ', '/quiet ', `
-    '/norestart ' -NoNewWindow -Wait;
+# Hash copied from https://github.com/slproweb/opensslhashes
+RUN $ExpectedHash = \"1e62e6459835f61c418532a1544e295be5400498b3569aadfb6d0b5d7a1c1dbc\";`
+    $ActualHash = $(Get-FileHash /local/Win64OpenSSL-1_1_1L.msi).Hash.ToLower();`
+    if ($ActualHash -ne $ExpectedHash) {`
+      throw \"OpenSSL hash mismatch. Expected: $ExpectedHash, Actual: $ActualHash\"`
+    };
+
+RUN Start-Process msiexec.exe -Wait -ArgumentList '/i C:\local\Win64OpenSSL-1_1_1L.msi /quiet';
```

Comment:
[Optional] Each `RUN` command adds a layer. Not a huge deal, but it might help to combine this one with the previous command…

Reply:
ooo you're right! tested it and it's a 9s diff! 

separate `RUN` commands
```
PS C:\Users\jschulz\ops-agent> Measure-Command { docker build -t testps -f .\dd . }

Days              : 0
Hours             : 0
Minutes           : 1
Seconds           : 14
Milliseconds      : 742
Ticks             : 747421059
TotalDays         : 0.000865070670138889
TotalHours        : 0.0207616960833333
TotalMinutes      : 1.245701765
TotalSeconds      : 74.7421059
TotalMilliseconds : 74742.1059
```

single `RUN` command
```
PS C:\Users\jschulz\ops-agent> Measure-Command { docker build -t testps -f .\dd . }

Days              : 0
Hours             : 0
Minutes           : 1
Seconds           : 5
Milliseconds      : 715
Ticks             : 657150537
TotalDays         : 0.000760590899305556
TotalHours        : 0.0182541815833333
TotalMinutes      : 1.095250895
TotalSeconds      : 65.7150537
TotalMilliseconds : 65715.0537
```